Qiskit for Quantum Computing (#6964)
section-engineering-education/engineering-education: content/articles/qiskit-for-quantum-computing/index.md

---
layout: engineering-education
status: publish
published: true
url: /qiskit-for-quantum-computing/
title: Qiskit for Quantum Computing
description: This tutorial aims to introduce our readers to Quantum computing. It also introduces them to the Qiskit toolkit and lets them experiment using Jupyter Notebooks hosted in IBM Quantum Lab.  
author: monica-dalmas
date: 2022-03-30T00:00:00-21:30
topics: [Networking]
excerpt_separator: <!--more-->
images:

  - url: /engineering-education/qiskit-for-quantum-computing/hero.png 
    alt: Qiskit for Quantum Computing Hero image
---
For most of us, it's hard to wrap our heads around the world of quantum computing. Most people have heard of it, but a lot of information about it glosses over important details. 
<!--more-->
This tutorial aims to clear this up and introduce you to Quantum computing. With the help of an open-source toolkit such as Qiskit, we'll introduce our readers to the toolkit and let them experiment using Jupyter Notebooks hosted in IBM Quantum Lab. 

### Quntum computers

> Quantum computers solve problems in a different way from the classic computers that we are familiar with. Quantum computers have several advantages over classical computers for certain problems. This is attributed to their ability to exist in a number of different states simultaneously. On the contrary, classical computers can only be in one state at a time.

Reviewer: ```suggestion
Quantum computers solve problems differently from the classic computers that we are familiar with. Quantum computers have several advantages over classical computers for certain problems. This is attributed to their ability to exist in several different states simultaneously. On the contrary, classical computers can only be in one state at a time.
```

Qiskit is written in Python. So, if you are proficient in the Python programming language, a lot of code will be familiar to you. The software framework is ideal for people who want to learn about quantum computing and get some actual hands-on experience. In terms of experimenting with quantum algorithms, you can run through examples of quantum circuits using their online tools.

### How to install Qiskit
To install qiskit, you have two options. You can install it locally on your local computer or on the Jupyter Notebook hosted by IBM. To install locally on a Linux machine, write the following code:

```bash
pip install -U pip && pip install qiskit
```
> This installation is for Linux machines. For Windows and macOS users, head on to the main website and select the appropriate machine you're using. A different installation code will be generated. Python version 3.6 and above is preferred.

> To get started online at IBM, we need to sign up [here](https://lab.quantum-computing.ibm.com) to get the API token which gives us permission to use their quantum devices. You can think of this are running Jupyter Notebook online with Qiskit installed.

Reviewer: ```suggestion
To get started online at IBM, we need to sign up [here](https://lab.quantum-computing.ibm.com) to get the API token that permits us to use their quantum devices. You can think of this are running Jupyter Notebook online with Qiskit installed.
```

> Remember your API token is your access code to IBM's quantum devices.

> You can get it by going to your `Profile` option on the top-right corner of the webpage and clicking the `Account details` option. In the API token section, you'll your API token hidden from view in asterics. Copy it and paste it inside the following code:

Reviewer: ```suggestion
You can get it by going to your `Profile` option on the top-right corner of the webpage and clicking the `Account details` option. In the API token section, you'll get your API token hidden from view in `***`. Copy it and paste it inside the following code:
```
